clowder-framework/clowder
# Extractor Catalog: Job History features

## Description
This PR adds a "Job History" view to the Extractor Catalog.

NOTE: It is still unclear whether this view will be eventually separated from the proposed "Metrics" view. 

### Problem
While we can view all extraction events for a given dataset or file, we do not have a way to view the entire job history for a given extractor.

### Approach
* Add the Extractor Job History view / route
* Link off to Job History view from Extractor Catalog
* Refactor existing job history view / grouping code to reuse as much as possible

### How to Test
Prerequisites: RabbitMQ plugin enabled, at least one extractor registered

1. Checkout and run this branch locally
2. Login to Clowder
3. Navigate to the Explore > Extractors view
4. Beside an extractor, expand the dropdown menu and choose "View History"
    * You should be taken to the new Job History view
    * The view should look very similar to the Extractions tab, displayed below files and datasets
    * You should see all jobs that have run by this extractor are listed here
    * You should be able to click a job to expand it and see the individual events from that job
5. Verify that the Extractions tab below files/datasets still appears and functions as intended

## Review Time Estimate


## Types of changes


## Checklist:
